index.js

```diff
@@ -70,13 +85,16 @@ function run(configPath, options, callback) {
         ));
     }
 
-    var maxRetries = parseInt(options.maxRetries, 10) || 0,
-        maxConcurrentWorkers = parseInt(options.maxConcurrentWorkers, 10)
-            || require('os').cpus().length,
-        workers = workerFarm({
-            maxRetries: maxRetries,
-            maxConcurrentWorkers: maxConcurrentWorkers
-        }, require.resolve('./src/webpackWorker'));
+    if (!validate(options)) {
+        return Promise.reject(new Error(
+          'Options validation failed:\n' +
+          validate.errors.map(function(error) {
+            return 'Property: "options' + error.dataPath + '" ' + error.message;
+          }).join('\n')
+        ));
+    }
+
+    var workers = workerFarm(options, require.resolve('./src/webpackWorker'));
```

Comment:
I'm fine with both as long as we keep compatibility with the current API. I'd like to avoid a 2.0 release until webpack 2.0 is out and we have full and well tested support for it to avoid confusion for users.

So I'm happy to merge either solution. You just need to pick the one that is more suitable for your use case. :)